EFAuditer/EFAuditServiceExtensions.cs

```diff
@@ -28,14 +28,15 @@ public static void AddEFAuditer(this IServiceCollection services, string connect
 
                 if (user != null)
                 {
-                    var userName = user.FindFirstValue(ClaimTypes.Upn);
-                    // Fallbacks if user doesn't have an email associated with them - as is the case with our test users
-                    if (string.IsNullOrEmpty(userName))
+                    // Fallback if user doesn't have an email associated with them - as is the case with our test users
+                    var userName = !string.IsNullOrEmpty(user.FindFirstValue(ClaimTypes.Upn))
+                        ? user.FindFirstValue(ClaimTypes.Upn)
+                        : user.Identity.Name;
+
+                    if (userName != null)
                     {
-                        userName = user.Identity.Name;
+                        scope.SetCustomField(CustomFields.AppUser, userName);

```

Comment:
As discussed, it's very surprising that setting the custom field here would override setting it directly on the context. It looks like this is already leading to bugs during notification import for drug resistance profiles and cluster data (which should show as system imported, but are actually showing the user). We should use a different custom field, `OverrideUser` when doing the override so that this can properly take precedence.